### DISCUSSION TITLE
feature: parallel should support limit Infinity and OnlyPositiveNumbers

### DISCUSSION THREAD

**Discussion by MarlonPassos-git** [radashi-org/radashi#235] [Ideas]
## Problem:

Currently, when we pass Infinite it throws an error
```ts
import * as _ from 'radashi'

const [errors, results] = await _.try(async () => {
  return _.parallel(Infinity, _.list(1, 3), async (num) => {
    await _.sleep(2000)

    return `hi_${num}`
  })
})()

console.log(errors, results)
```
![image](https://github.com/user-attachments/assets/294ff48f-3114-47c0-9786-3501d4d7e9df)

It's easy to fix it with. 
```ts
const queues = list(1, Math.min(limit, array.length)).map(() => new Promise(processor));
```

This can also be good because if the person passes a value in parallels that is greater than the size of the array, we don't create unnecessary things.

but testing this I ended up realizing that if I pass 0 or a negative number it doesn't work either.

```ts
import * as _ from 'radashi'

const [errors, results] = await _.try(async () => {
  return _.parallel(-100, _.list(1, 3), async (num) => {
    await _.sleep(2000)

    return `hi_${num}`
  })
})()

console.log(errors, results)
```
![image](https://github.com/user-attachments/assets/cb311ec4-32a7-46a5-bf6a-32eae2c71d3b)

I think it would be interesting to deal with this case, in the code we could validate that the number has to be greater than 0
```ts
if (limit < 1)  {
  Throw new Error ("Only positive numbers are allowed.")
}
```
and on the typescript side I think we could use this type to enforce positive values with this type 

```ts
type PositiveInteger<T extends number> =
  `${T}` extends '0' | `-${any}` | `${any}.${any}` ? NegativeNumber : T
```

![image](https://github.com/user-attachments/assets/6072e626-3cdd-4fb4-b2db-95489934412f)

However, in this case, I have doubts whether this would be a major change. 🤔

## Related Issues 

https://github.com/sodiray/radash/issues/426

**Comment by aleclarson** [comment]
Definitely agree on Infinity.

I'd be fine with clamping anything less than 1 to be equal to 1.

**Reply by MarlonPassos-git** [reply]
ok i wiil work on it. About the type `PositiveInteger`, what do you whink ?

**Reply by aleclarson** [reply]
It's a cool idea, but we should wait on it until we can give a better type-level error message (tracked by [this TypeScript issue](https://github.com/microsoft/TypeScript/issues/23689)). Besides, it's unlikely that anyone will type 0 or a negative number in there, IMO.

**Comment by MarlonPassos-git** [comment]
https://github.com/radashi-org/radashi/pull/238
